spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8BootstrapConfiguration.java

```diff
@@ -39,31 +40,24 @@
  * @author Ioannis Canellos
  */
 @Configuration(proxyBeanMethods = false)
-@ConditionalOnProperty(value = "spring.cloud.kubernetes.enabled", matchIfMissing = true)
+@ConditionalOnKubernetesEnabled
+@Import({ KubernetesCommonsAutoConfiguration.class, Fabric8AutoConfiguration.class })
 @ConditionalOnClass({ ConfigMap.class, Secret.class })
 @AutoConfigureAfter(KubernetesBootstrapConfiguration.class)
 public class Fabric8BootstrapConfiguration {
 
-	@Configuration(proxyBeanMethods = false)
-	@Import({ KubernetesCommonsAutoConfiguration.class, Fabric8AutoConfiguration.class })
-	protected static class KubernetesPropertySourceConfiguration {
-
-		@Autowired
-		private KubernetesClient client;
-
-		@Bean
-		@ConditionalOnProperty(name = "spring.cloud.kubernetes.config.enabled", matchIfMissing = true)
-		public Fabric8ConfigMapPropertySourceLocator configMapPropertySourceLocator(
```

Comment:
remove the inner class. initially I though that it is needed because of the `@Import({ KubernetesCommonsAutoConfiguration.class, Fabric8AutoConfiguration.class })`. So when `spring.cloud.kubernetes.enabled=false` these would not be imported but:

 - `KubernetesCommonsAutoConfiguration` and `Fabric8AutoConfiguration` are annotated with  `@ConditionalOnKubernetesEnabled`, so that is not a problem
 - even if I add a dummy config (without any annotations), import it, it will still not be picked up (the constructor inside it is not even called) when `spring.cloud.kubernetes.enabled=false`.

As such, this looks like a safe change to me. I hope it makes sense